feature-find-meal/src/main/java/com/kinandcarta/feature/find/meal/view/FindMealFragment.kt

```diff
@@ -10,106 +8,101 @@ import android.widget.FrameLayout
                 progressBarView.isVisible = true
-            is FindMealViewModel.State.LoadedMeals ->
+            is DisplayMealsViewModel.State.LoadedMeals ->
                 onLoadedMeals(state)
-            is FindMealViewModel.State.ReservedMeal ->
+            is DisplayMealsViewModel.State.ReservedMeal ->
                 onReservedMeal(state)
-            is FindMealViewModel.State.MealUnavailable ->
+            is DisplayMealsViewModel.State.MealUnavailable ->
                 onMealUnavailable(state)
-            is FindMealViewModel.State.Failed ->
+            is DisplayMealsViewModel.State.LocationUpdate ->
+                onLocationUpdate(state)
+            is DisplayMealsViewModel.State.Failed ->
                 onFailure(state.failure)
         }
     }
 
-    private fun onLoadedMeals(state: FindMealViewModel.State.LoadedMeals) {
+    private fun onLoadedMeals(state: DisplayMealsViewModel.State.LoadedMeals) {
         progressBarView.isVisible = false
         mealsAdapter.submitList(state.meals)
     }
 
-    private fun onReservedMeal(state: FindMealViewModel.State.ReservedMeal) {
-        showToast("Your meal reservation code is ${state.code}")
+    private fun onReservedMeal(state: DisplayMealsViewModel.State.ReservedMeal) {
+        viewModel.updateMeals(userLatLng)
+        showToast("${getString(R.string.reservation_message)} ${state.code}")
+    }
+
+    private fun onMealUnavailable(state: DisplayMealsViewModel.State.MealUnavailable) {
+        showToast(getString(R.string.meal_unavailable_message))
     }
 
-    private fun onMealUnavailable(state: FindMealViewModel.State.MealUnavailable) {
-        showToast("Unfortunately this meal is unavailable")
+    private fun onLocationUpdate(state: DisplayMealsViewModel.State.LocationUpdate) {
+        userLatLng = state.userLatLng
+        viewModel.updateMeals(userLatLng)
     }
 
-    private fun onFailure(failure: FindMealViewModel.Failure) {
+    private fun onFailure(failure: DisplayMealsViewModel.Failure) {
         when (failure) {
-            is FindMealViewModel.Failure.LoadingMealsFailed -> {
+            is DisplayMealsViewModel.Failure.LoadingMealsFailed -> {
                 progressBarView.isVisible = false
-                showToast(failure.localizedMessage ?: "An unexpected error occurred loading meals")
+                showToast(failure.localizedMessage ?: getString(R.string.error_loading_meals))
             }
-            is FindMealViewModel.Failure.ReserveAMealFailed -> {
-                showToast(failure.localizedMessage ?: "An unexpected error occurred reserving a meal")
+            is DisplayMealsViewModel.Failure.ReserveAMealFailed -> {
+                showToast(
+                    failure.localizedMessage ?: getString(R.string.error_reserving_meal)
+                )
             }
         }
     }
 
     companion object {
-        private const val LOCATION_PERMISSION_REQUEST_CODE = 1
+        val newInstance = FindMealFragment()
```

Comment:
```suggestion
        fun newInstance() = FindMealFragment()
```
This will create a new instance every time you need to use it from anywhere. Otherwise, you'll only have ONE instance